### PR TITLE
[trivial] Add end of namespace comments. Improve consistency.

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

**Comment by practicalswift** [comment]
Merge conflict resolved!

File: src/base58.cpp [hunk 1]
```diff
@@ -225,7 +225,7 @@ class CBitcoinAddressVisitor : public boost::static_visitor<bool>
     bool operator()(const CNoDestination& no) const { return false; }
 };
 
-} // anon namespace
+} // namespace
```
Comment (jtimon) [inline src/base58.cpp]: why remove anon ?

Comment (practicalswift) [inline src/base58.cpp]: @jtimon My reasoning behind that was to achieve consistency how anonymous namespaces are indicated. Consistency in order to improve readability and allow for automatic identification of missing/incorrect end of namespace comments.

`// namespace` is the form preferred by the clang-tooling (`clang-tidy`, etc).

Let me know if the choice of `// anon namespace` is intentional and I'll adjust this PR :-)

Comment (jtimon) [inline src/base58.cpp]: Well, it just seems to me that the same consistency can be achieved leaving "anon namespace" for anonymous namespaces with less disruption. But if clang prefers ```// namespace``` or ```// anonymous namespace``` I don't have a strong opinion.

**Comment by practicalswift** [comment]
Any changes needed before merge? :-)

**Comment by practicalswift** [comment]
@paveljanik I see! If the consensus opinion is that end of namespace marks are optional then this PR is redundant and I'll close it :-)

**Comment by laanwj** [comment]
I think the point is that if end-of-namespace comments are to be required, they should be mentioned in `doc/developer-notes.md` along with rationale. It should be clear to contributors what the "rules" are. If not, changes like this come out of the blue.

**Comment by paveljanik** [comment]
@practicalswift I do now know what the consensus is, but my opinion is the same as @laanwj 's.

**Comment by practicalswift** [comment]
@laanwj Good point! I've added a note to `doc/developer-notes.md` in this PR.

**Comment by paveljanik** [comment]
I'd slightly prefer `// anonymous namespace` over `// namespace` at the end of anonymous namespace, but this is minor.

**Comment by practicalswift** [comment]
@paveljanik It seems like `// namespace` is the dominant form currently used in the repo:

```
$ git checkout master
$ git grep -E '} *// namespace$' | wc -l
21
$ git grep -E '} *// anonymous namespace$' | wc -l
3
```

`// namespace` is also the form preferred by the clang-tooling (`clang-tidy`, etc).

**Comment by paveljanik** [comment]
@practicalswift yes. But it seems like the author forget to add the real name of the namespace after it ;-)

File: src/univalue/lib/univalue.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ bool ParseDouble(const std::string& str, double *out)
     if(out) *out = result;
     return text.eof() && !text.fail();
 }
-}
+} // namespace
```
Comment (paveljanik) [inline src/univalue/lib/univalue.cpp]: This is univalue -> please report upstream.

**Comment by paveljanik** [comment]
Concept ACK.

Please remove univalue change. Then utACK

**Comment by practicalswift** [comment]
@paveljanik Removed univalue change as requested :-)

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9544/commits/85306068d6cd171519d18b3425d67fc0d5ebad99

Thanks!

**Comment by practicalswift** [comment]
@paveljanik Thanks for reviewing! Let me know if any further tweaks are needed :-)

**Comment by practicalswift** [comment]
Conflicts resolved!

**Comment by jtimon** [comment]
"Fast re-review checked that still doesn't need testing ACK" 486a5d7

File: doc/developer-notes.md [hunk 3]
```diff
@@ -408,6 +408,21 @@ Source code organization
 
   - *Rationale*: Avoids symbol conflicts
 
+- Terminate namespaces with a comment (`// namespace mynamespace`). The comment
+  should be placed on the same line as the brace closing the namespace, e.g.
+
+  ```c++
```
Comment (jtimon) [inline doc/developer-notes.md]: nit: extra spaces?

Comment (practicalswift) [inline doc/developer-notes.md]: @jtimon Sorry I'm not sure I follow - where should the extra spaces be placed? :-)

Comment (jtimon) [inline doc/developer-notes.md]: s/  ```/```/ ?

**Comment by practicalswift** [comment]
@jtimon Fixed! :-)

**Comment by laanwj** [comment]
utACK 5a9b508

**Pull request merged**